src/openms/source/FORMAT/MRMFeaturePickerFile.cpp

```diff
@@ -0,0 +1,160 @@
+
+#include <OpenMS/FORMAT/MRMFeaturePickerFile.h>
+
+namespace OpenMS
+{
+  void MRMFeaturePickerFile::load(
+    const String& filename,
+    std::vector<MRMFeaturePicker::ComponentParams>& cp_list,
+    std::vector<MRMFeaturePicker::ComponentGroupParams>& cgp_list
+  )
+  {
+    cp_list.clear();
+    cgp_list.clear();
+    CsvFile::load(filename, ',', false);
+    StringList sl;
+    std::map<String, Size> headers;
+    if (rowCount() >= 2) // no need to read headers if that's the only line inside the file
+    {
+      getRow(0, sl);
+      for (Size i = 0; i < sl.size(); ++i)
+      {
+        headers[sl[i]] = i; // for each header found, assign an index value to it
+      }
+      if (!(headers.count("component_name") && headers.count("component_group_name")))
+      {
+        throw Exception::MissingInformation(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "Columns component_name and/or component_group_name not found.");
+      }
+    }
+    for (Size i = 1; i < rowCount(); ++i)
+    {
+      getRow(i, sl);
+      MRMFeaturePicker::ComponentParams cp;
+      MRMFeaturePicker::ComponentGroupParams cgp;
+      if (extractParamsFromLine_(sl, headers, cp, cgp))
+      {
+        cp_list.push_back(cp);
+        // The following lines check if cgp is already present in cgp_list
+        // If that is not the case, the extracted cgp is pushed
+        bool cgp_found = std::any_of(
+          cgp_list.begin(),
+          cgp_list.end(),
+          [&cgp](MRMFeaturePicker::ComponentGroupParams current_cgp)
+          {
+            return cgp.component_group_name == current_cgp.component_group_name;
+          }
+        );
+        if (!cgp_found)
+        {
+          cgp_list.push_back(cgp);
+        }
+      }
+    }
+  }
+
+  bool MRMFeaturePickerFile::extractParamsFromLine_(
+    const StringList& line,
+    const std::map<String, Size>& headers,
+    MRMFeaturePicker::ComponentParams& cp,
+    MRMFeaturePicker::ComponentGroupParams& cgp
+  ) const
+  {
+    cp.component_name = line[headers.find("component_name")->second]; // find the component_name value
+    cp.component_group_name = line[headers.find("component_group_name")->second]; // find the component_group_name value
+    if (cp.component_name == "" || cp.component_group_name == "") // TODO: should it skip on empty group name?
+    {
+      return false;
+    }
+    cgp.component_group_name = cp.component_group_name; // save the component_group_name also into cgp
+    for (const std::pair<String, Size>& h : headers) // parse the parameters
+    {
+      const String& header = h.first;
+      const Size& i = h.second;
+      boost::smatch m;
+      if (boost::regex_search(header, m, boost::regex("TransitionGroupPicker:(?!PeakPickerMRM:)(.+)")))
+      {
+        setCastValue(String(m[1]), line[i], cgp.params);
+      }
+      else if (boost::regex_search(header, m, boost::regex("TransitionGroupPicker:PeakPickerMRM:(.+)")))
```

Comment:
This looks to be reversed:  the transitionGroupPicker operates on the transition groups while PeakPickerMRM operates on the individual transitions.

Reply:
That's what that code is doing: if the regex is NOT followed by "PeakPickerMRM", then the info is inserted in `cgp`.
Otherwise, in `cp`

Reply:
OK gotcha.

Reply:
Unfortunately, I am not that familiar with the boost implementation of regex.